File: modules/crypto-core/src/main/java/com/shterneregen/securelan/crypto/workflow/PasswordFileCryptoWorkflow.java
package com.shterneregen.securelan.crypto.workflow;

import com.shterneregen.securelan.crypto.model.PasswordEncryptedData;

public interface PasswordFileCryptoWorkflow {
    PasswordEncryptedData encrypt(byte[] fileBytes, char[] password);

    byte[] decrypt(PasswordEncryptedData encryptedData, char[] password);
}

File: modules/crypto-core/src/main/kotlin/com/shterneregen/securelan/crypto/workflow/PasswordFileCryptoWorkflow.kt
package com.shterneregen.securelan.crypto.workflow

import com.shterneregen.securelan.crypto.model.PasswordEncryptedData

interface PasswordFileCryptoWorkflow {
    fun encrypt(fileBytes: ByteArray, password: CharArray): PasswordEncryptedData

    fun decrypt(encryptedData: PasswordEncryptedData, password: CharArray): ByteArray
}
